src/VisualizationTests/HelixWatch3dViewModelPrimitiveTests.cs

```diff
@@ -0,0 +1,533 @@
+            pointColors = points.SelectMany(x => x.Geometry.Colors);
+            //isolate sets Alpha to .3 currently
+            var isolatedColors = pointColors.Where(x => x.Alpha < .4f).Count();
+            Assert.AreEqual(29791, isolatedColors);
+
+            //select points
+            DynamoSelection.Instance.ClearSelection();
+            DynamoSelection.Instance.Selection.Add(pointDisplayNode);
+            DispatcherUtil.DoEvents();
+            //assert points are no longer isolated - but they should be frozen.
+            pointColors = points.SelectMany(x => x.Geometry.Colors);
+            frozenColorsCount = pointColors.Where(x => x.Alpha == .5).Count();
+            Assert.AreEqual(29791, frozenColorsCount);
+
+            DynamoSelection.Instance.ClearSelection();
+            DispatcherUtil.DoEvents();
+            //isolated again.
+            pointColors = points.SelectMany(x => x.Geometry.Colors);
+            //isolate sets Alpha to .3 currently
+            isolatedColors = pointColors.Where(x => x.Alpha < .4f).Count();
+            Assert.AreEqual(29791, isolatedColors);
+
+            //unisolate
+            ViewModel.BackgroundPreviewViewModel.IsolationMode = false;
+            DispatcherUtil.DoEvents();
+            pointColors = points.SelectMany(x => x.Geometry.Colors);
+            frozenColorsCount = pointColors.Where(x => x.Alpha == .5).Count();
+            Assert.AreEqual(29791, frozenColorsCount);
+
+            //back to normal.
+            pointDisplayNode.IsFrozen = false;
+            DispatcherUtil.DoEvents();
+            pointColors = points.SelectMany(x => x.Geometry.Colors);
+            bluePtsCount = pointColors.Where(x => x == Colors.Blue.ToColor4()).Count(); ;
+            yellowPtsCount = pointColors.Where(x => x == new Color3(1, 1, 0).ToColor4()).Count();
+            Assert.AreEqual(14896, bluePtsCount);
+            Assert.AreEqual(14895, yellowPtsCount);
+        }
+
+        #endregion
+
+        #region special renderpackages
+        [Test]
+        public void RenderCoreHasFlagsSetCorrectlyForSpecialRenderPackage()
+        {
+            var ws = ViewModel.Model.CurrentWorkspace as HomeWorkspaceModel;
+            var mockManipulator = new DynamoGeometryModel3D()
+            {
+                Transform = MatrixTransform3D.Identity,
+                Material = PhongMaterials.Red,
+                IsHitTestVisible = true,
+                RequiresPerVertexColoration = false,
+            };
+            Assert.False((mockManipulator.SceneNode.RenderCore as DynamoGeometryMeshCore).IsSpecialRenderPackageData);
+            AttachedProperties.SetIsSpecialRenderPackage(mockManipulator, true);
+            //assert that setting this attached property updated the meshcore data.
+            Assert.IsTrue((mockManipulator.SceneNode.RenderCore as DynamoGeometryMeshCore).IsSpecialRenderPackageData);
+          
+        }
+        #endregion
+
+    }
+}
```

Comment:
NIT: Let's use `PascalCase` (that means no `_` separator) in method names.

Reply:
Updated